java/vector/src/main/java/org/apache/arrow/vector/file/ArrowMagic.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.arrow.vector.file;
+
+import java.nio.charset.StandardCharsets;
+
+public class ArrowMagic {
+  protected static final byte[] MAGIC = "ARROW1".getBytes(StandardCharsets.UTF_8);
```

Comment:
since a byte array is mutable consider making it private and adding methods here related to MAGIC number validation:
```
int getMagicLength() {
  return MAGIC.length
}
void validateMagicNumber(byte[] array) {
  if (!Arrays.equals(MAGIC, array)) {
    throw new InvalidArrowFileException("missing Magic number " + Arrays.toString(buffer.array()));		
  }		
}
```
(taken from ArrowFileReader)